### PR TITLE
Javascript javascript3 week2

### PR DESCRIPTION
**Pull request by Impressiv3**

### CODE REVIEW

File: javascript/javascript3/week2/Task1_Movies_Exercise/movies.exercise.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+///////////////////////////////////////////////////////////////////////////////
+
+async function getData(url) {
+  let response = await fetch(url);
+  if (response.status >= 200 && response.status < 300) {
+    let data = response.json();
+    return data;
+  } else {
+    throw new Error(response.status);
```
Comment (florinmaticu) [inline javascript/javascript3/week2/Task1_Movies_Exercise/movies.exercise.js:29]: Argument of type 'number' is not assignable to parameter of type 'string | undefined'.ts(2345)

File: javascript/javascript3/week2/Task3_Rewrite_Time/rewrite_time.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ As long as you can use it as a promise. */
+
+const setTimeoutPromise = (delay) => new Promise((resolve) => setTimeout(resolve, delay));
+
+setTimeoutPromise(3000).then(() => {
+  console.log("Called after 3 seconds");
+});
+
+const getCurrentLocation = () => {
+  return new Promise((resolve, reject) => {
```
Comment (florinmaticu) [inline javascript/javascript3/week2/Task3_Rewrite_Time/rewrite_time.js:14]: You can try adding code for the case when the request fails and therefore you can call `reject`

File: javascript/javascript3/week2/Task4_Fetching_Waiting/fetching_waiting.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+// TASK 4 Fetching and Waiting
+/*  The 3 steps:
+    Wait 3 seconds
+    After 3 seconds fetch some data from any api you like
+    Log out the data from the api */
+
+// USING PROMISES AND .then
+
+url = "https://api.chucknorris.io/jokes/random";
```
Comment (florinmaticu) [inline javascript/javascript3/week2/Task4_Fetching_Waiting/fetching_waiting.js:9]: `const url = "https://api.chucknorris.io/jokes/random";`

File: javascript/javascript3/week2/Task4_Fetching_Waiting/fetching_waiting.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+function delayFetch() {
+  setTimeout(() => {
+    fetch(url)
+      .then((response) => response.json())
+      .then((data) => {
+        console.log(data.value);
+      }, 3000)
```
Comment (florinmaticu) [inline javascript/javascript3/week2/Task4_Fetching_Waiting/fetching_waiting.js:17]: There is a compile time error on line 17.
```
Argument of type '3000' is not assignable to parameter of type '((reason: any) => PromiseLike<never>) | null | undefined'.ts(2345.
```

File: javascript/javascript3/week2/Task1_Movies_Exercise/movies.exercise.js (hunk 1 above)
Comment (Infamou5) [reply javascript/javascript3/week2/Task1_Movies_Exercise/movies.exercise.js:29]: DeleteMe1

Comment (Infamou5) [reply javascript/javascript3/week2/Task1_Movies_Exercise/movies.exercise.js:29]: IF I run the code on my machine it works without any errors. I have no clue how to fix this problem.
-> this has to be the line that is giving you an error : if (response.status >= 200 && response.status < 300.
Open for suggestions.

File: javascript/javascript3/week2/Task3_Rewrite_Time/rewrite_time.js (hunk 2 above)
Comment (Infamou5) [reply javascript/javascript3/week2/Task3_Rewrite_Time/rewrite_time.js:14]: Reject case has been added to the getCurrentLocation function.

File: javascript/javascript3/week2/Task4_Fetching_Waiting/fetching_waiting.js (hunk 3 above)
Comment (Infamou5) [reply javascript/javascript3/week2/Task4_Fetching_Waiting/fetching_waiting.js]: Yes, you are damn right! I had this code together with the other folders so const url was already defined somewhere else. 
I have changed it to const url now.

File: javascript/javascript3/week2/Task4_Fetching_Waiting/fetching_waiting.js (hunk 4 above)
Comment (Infamou5) [reply javascript/javascript3/week2/Task4_Fetching_Waiting/fetching_waiting.js:17]: I have moved the delay time of 3000 to its place so now the error you are getting should be fixed.

**Review by flom84 — APPROVED** [review]

**Pull request merged**